test/D2L.Services.Core.Extensions.UnitTests/ParallelizationExtensionsTests.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using D2L.Services;
+using NUnit.Framework;
+
+namespace D2L.Services {
+	
+	[TestFixture]
+	[Category( "Unit" )]
+	internal sealed class ParallelizationExtensionsTests {
+		
+		private Random m_rng;
+		
+		[TestFixtureSetUp]
+		public void TestFixtureSetUp() {
+			m_rng = new Random( DateTime.Now.GetHashCode() );
+		}
+		
+		[Test]
+		public async Task FilterInParallelAsyncTest() {
+			int[] numbers = new[]{ 1, 6, 1, 8, 0, 3, 3, 9, 8, 8, 7, 5 };
+			int[] oddOnly = new[]{ 1,    1,       3, 3, 9,       7, 5 };
+			
+			IEnumerable<int> result = await numbers.FilterInParallelAsync( async n => {
+				await RandomDelay( 900 ).SafeAsync();
+				return n % 2 != 0;
+			}).SafeAsync();
+			
+			CollectionAssert.AreEqual( oddOnly, result );
+		}
+		
+		[Test]
+		public async Task MapInParallelAsyncTest() {
+			int[] numbers = new[]{ 3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9 };

```

Comment:
Consider simplifying the numbers and math here.  Wouldn't an array of ascending int, where you just add 1 to each element suffice?